plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -60,23 +90,30 @@ protected void setup(Binder binder)
         newOptionalBinder(binder, Key.get(UserNameProvider.class, ForHiveMetastore.class))
                 .setDefault()
                 .toInstance(SIMPLE_USER_NAME_PROVIDER);
-
         binder.bind(Key.get(boolean.class, AllowHiveTableRename.class)).toInstance(true);
-
-        install(new ThriftMetastoreAuthenticationModule());
-    }
-
-    @Provides
-    @Singleton
-    @ThriftHiveWriteStatisticsExecutor
-    public ExecutorService createWriteStatisticsExecutor(ThriftMetastoreConfig hiveConfig)
-    {
-        return newFixedThreadPool(hiveConfig.getWriteStatisticsThreads(), threadsNamed("hive-thrift-statistics-write-%s"));
     }
 
     @PreDestroy
     public void shutdownsWriteStatisticExecutor(@ThriftHiveWriteStatisticsExecutor ExecutorService executor)
     {
         executor.shutdownNow();
     }
+
+    private static class ThriftHiveMetastoreStatisticExecutorProvider
+            implements Provider<ExecutorService>
+    {
+        private final ThriftMetastoreConfig thriftMetastoreConfig;
```

Comment:
Please don't put mutable config class on fields. This should be `int writeStatisticsThreads`.

Reply:
I thought we discussed this in https://github.com/trinodb/trino/pull/20371#discussion_r1487149670 and the suggestion was to use a static class instead of record. If I try to pass in int, the binder errors out as I mentioned in that comment previously.

Reply:
No need to change the constructor's argument.

Reply:
Ah I see what you mean now. Let me change it.

Reply:
done.